grails-app/services/org/broadinstitute/orsp/DataUseLetterService.groovy

```diff
@@ -0,0 +1,48 @@
+package org.broadinstitute.orsp
+
+import grails.gorm.transactions.Transactional
+
+class DataUseLetterService {
+    QueryService queryService
+
+    String generateDul(Object input) {
+        DataUseLetter newDul = new DataUseLetter()
+
+        String uniqueId = UUID.randomUUID().toString()
+
+        newDul.setUid(uniqueId)
+        newDul.setCreator(input["user"] as String)
+
+        newDul.setConsentGroupKey(input["consentKey"] as String)
+        newDul.setSubmitted(false)
+        newDul.setCreationDate(new Date())
+
+        newDul.save(flush: true)
+
+        if (newDul.hasErrors()) {
+            throw new DomainException(newDul.getErrors())
+        } else {
+            newDul.uid
+        }
+    }
+
+    @Transactional
+    def udpateDataUseLetter(Object input) {
```

Comment:
Same feedback as above. You can avoid casting `input.dulInfo as String` if there was a proper object here to refer to.